test/misc/unit/libraries/ConversionLib.t.sol

```diff
@@ -0,0 +1,176 @@
+        D18 pricePoolPerAsset = d18(uint128(bound(pricePoolPerAsset_, MIN_PRICE, MAX_PRICE)));
+        D18 pricePoolPerShare = d18(uint128(bound(pricePoolPerShare_, MIN_PRICE, MAX_PRICE)));
+
+        uint256 poolAmount =
+            ConversionLib.convertWithPrice(assetAmount, assetDecimals, SHARE_DECIMALS, pricePoolPerAsset);
+        uint256 expectedShareAmount = pricePoolPerShare.reciprocalMulUint256(poolAmount);
+
+        uint256 result = ConversionLib.assetToShareAmount(
+            assetAmount, assetDecimals, SHARE_DECIMALS, pricePoolPerAsset, pricePoolPerShare
+        );
+
+        assertEq(result, expectedShareAmount, "assetToShareAmount failed");
+    }
+
+    function testAssetToShareToAssetRoundTrip(
+        uint128 assetAmount,
+        uint8 assetDecimals,
+        uint64 pricePoolPerAsset_,
+        uint64 pricePoolPerShare_
+    ) public pure {
+        assetDecimals = uint8(bound(assetDecimals, MIN_ASSET_DECIMALS, MAX_ASSET_DECIMALS));
+        assetAmount = uint128(bound(assetAmount, 10 ** assetDecimals, MAX_AMOUNT));
+        D18 pricePoolPerAsset = d18(uint128(bound(pricePoolPerAsset_, MIN_PRICE, MAX_PRICE)));
+        D18 pricePoolPerShare = d18(uint128(bound(pricePoolPerShare_, MIN_PRICE, MAX_PRICE)));
+
+        uint256 shareAmount = ConversionLib.assetToShareAmount(
+            assetAmount, assetDecimals, POOL_DECIMALS, pricePoolPerAsset, pricePoolPerShare
+        );
+        uint256 assetRoundTrip = ConversionLib.shareToAssetAmount(
+            shareAmount, POOL_DECIMALS, assetDecimals, pricePoolPerShare, pricePoolPerAsset
+        );
+
+        assertApproxEqAbs(
+            assetRoundTrip, assetAmount, MIN_PRICE * 10, "Asset->Share->Asset roundtrip target precision excess"
+        );
+    }
+
+    /// NOTE: Solely serves to represent the horrible precision for this round trip due to reciprocal multiplication
+    function testShareToAssetToShareRoundTrip(
+        uint128 shareAmount,
+        uint8 assetDecimals,
+        uint64 pricePoolPerAsset_,
+        uint64 pricePoolPerShare_
+    ) public pure {
+        assetDecimals = uint8(bound(assetDecimals, 6, MAX_ASSET_DECIMALS));
+        D18 pricePoolPerAsset = d18(uint128(bound(pricePoolPerAsset_, MIN_PRICE, MAX_PRICE)));
+        D18 pricePoolPerShare = d18(uint128(bound(pricePoolPerShare_, 1e14, MAX_PRICE)));
+        shareAmount = uint128(bound(shareAmount, 10 ** assetDecimals, type(uint128).max / pricePoolPerShare.inner()));
+
+        uint256 assetAmount = ConversionLib.shareToAssetAmount(
+            shareAmount, POOL_DECIMALS, assetDecimals, pricePoolPerShare, pricePoolPerAsset
+        );
+        uint256 shareRoundTrip = ConversionLib.assetToShareAmount(
+            assetAmount, assetDecimals, POOL_DECIMALS, pricePoolPerAsset, pricePoolPerShare
+        );
+        assertApproxEqAbs(
+            shareRoundTrip, shareAmount, MAX_PRICE, "Share->Asset->Share roundtrip target precision excess"
+        );
```

Comment:
NOTE: I spend quite some time on this test and the calculations of `ConversionLib` because this precision delta of 1e20 is just horrible for the roundtrip of assetAmount -> shares -> assetAmount. The other roundtrip works much better with 1e11 precision (still bad). I wanted to keep this test mainly for future reference.